src/js_realm.hpp

```diff
@@ -987,6 +989,60 @@ void RealmClass<T>::compact(ContextType ctx, ObjectType this_object, Arguments a
 }
 
 #if REALM_ENABLE_SYNC
+namespace {
+
+// FIXME: Sync should provide this: https://github.com/realm/realm-sync/issues/1796
+inline sync::ObjectID object_id_from_string(std::string const& string)
+{
+    if (string.front() != '{' || string.back() != '}')
+        throw std::invalid_argument("Invalid object ID.");
+
+    size_t dash_index = string.find('-');
+    if (dash_index == std::string::npos)
+        throw std::invalid_argument("Invalid object ID.");
+
+    std::string high_string = string.substr(1, dash_index - 1);
+    std::string low_string = string.substr(dash_index + 1, string.size() - dash_index - 2);
+
+    if (high_string.size() == 0 || high_string.size() > 16 || low_string.size() == 0 || low_string.size() > 16)
+        throw std::invalid_argument("Invalid object ID.");
+
+    auto isxdigit = static_cast<int(*)(int)>(std::isxdigit);
+    if (!std::all_of(high_string.begin(), high_string.end(), isxdigit) ||
+        !std::all_of(low_string.begin(), low_string.end(), isxdigit)) {
+        throw std::invalid_argument("Invalid object ID.");
+    }
+
+    return sync::ObjectID(std::stoi(high_string, nullptr, 16), std::stoi(low_string, nullptr, 16));
```

Comment:
`stoi` returns an `int`, but each part of the object ID is up to 64 bits (unsigned) — use `std::stoull` instead. :-)

Any object with a string primary key is likely to have these upper bits set.

Reply:
If only sync provided a function that handles this correctly so I don't have to implement it poorly myself 😄 (https://github.com/realm/realm-sync/issues/1796)

Reply:
Acknowledeged ;)

Shuffling this around is one of my motivations for reversing the dependency between Object Store and Sync. Particularly, I want to put the "Object ID" concept into Object Store, so everything is at least in one place.